programming-extensions/programming-extension-processbuilder/src/main/java/org/objectweb/proactive/extensions/processbuilder/LinuxProcessBuilder.java

```diff
@@ -663,33 +664,60 @@ public OSLinuxProcess(Process p, OSUser user, String token, String scriptBaseFol
 
         @Override
         public void destroy() {
+            Process p = null;
             try {
                 LinuxProcessBuilder lpb = new LinuxProcessBuilder(user, null, scriptBaseFolder);
                 String killProcessTreeScript = new File(scriptBaseFolder,
                                                         PROCESS_TREE_KILLER_LOCATION).getAbsolutePath();
                 lpb.directory(new File(System.getProperty("java.io.tmpdir")));
                 lpb.command(killProcessTreeScript, this.cleanupTimeoutGetter.getCleanupTimeSecondsString());
                 lpb.environment().put("PROCESS_KILL_TOKEN", this.token);
-                Process p = lpb.start();
+                p = lpb.start();
                 p.waitFor();
 
                 if (logger.isDebugEnabled()) {
-                    BufferedReader br;
+                    printStdout(p, Level.DEBUG);
+                    printStderr(p, Level.DEBUG);
+                    logger.debug("token: " + this.token + " exitval:" + p.exitValue());
+                }
+            } catch (Exception e) {
+                printStderr(p, Level.WARN);
+                logger.warn("Failed to destroy OS process with token: " + this.token, e);
+            }
+
+        }
+
+        private void printStderr(Process p, Level logLevel) {
+            if (p == null) {
+                return;
+            }
+            printStream(p.getErrorStream(), "stderr", logLevel);
+
+        }
 
-                    br = new BufferedReader(new InputStreamReader(p.getInputStream()));
-                    for (String line = br.readLine(); line != null; line = br.readLine()) {
-                        logger.debug("token: " + this.token + " stdout:" + line);
+        private void printStdout(Process p, Level logLevel) {
+            if (p == null) {
+                return;
+            }
+            printStream(p.getInputStream(), "stdout", logLevel);
+        }
+
+        private void printStream(InputStream inputStream, String messageType, Level logLevel) {
+
+            try {
+                BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+                for (String line = br.readLine(); line != null; line = br.readLine()) {
+                    if (Level.DEBUG == logLevel) {
```

Comment:
better to do this in a single line
logger.log(logLevel, "token: " + this.token + " " + messageType + ":" + line)